JASP-Desktop/qquick/datasetview.cpp

```diff
@@ -351,6 +332,13 @@ void DataSetView::addLine(float x0, float y0, float x1, float y1)
 	_lines[_linesActualSize++] = y1;
 }
 
+QSizeF DataSetView::getTextSize(const QString & text) const
+{
+	QFontMetricsF metricsFont(JaspTheme::currentTheme()->font());
```

Comment:
The QFontMetricsF should be a property of the currentTheme: it should not be created each time it is needed, but only when the font is changed.

Reply:
Yeah, but the problem is that I'm not entirely sure if it responds properly when the `pixelSize` gets set in `Theme.qml`.
And I also don't really know how to check it.

If you look at: https://github.com/jasp-stats/jasp-desktop/blob/8a6fdb171a816bbba653b1943e8e016399504db6/JASP-Desktop/components/JASP/Theme/Theme.qml#L29
Does a change to uiScale get propagated? 

Although I guess I could just reset the stored QFontMetricsF whenever uiScale changes?
Because that wouldn't be very different from how I do it there right?
Yeah I can change it to something like that.

Reply:
There is maybe a concurrency problem here (I love concurrency problem): if uiScale is changed, the data view change can be first called before the fontmetrics is changed...

Reply:
good point, check the latest commit.